vault/resource_azure_secrets_sync_destination.go

```diff
@@ -0,0 +1,174 @@
+				Description: "Unique name of the Azure destination.",
+				ForceNew:    true,
+			},
+			fieldKeyVaultURI: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "URI of an existing Azure Key Vault instance.",
+				ForceNew:    true,
+			},
+			consts.FieldClientID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "Client ID of an Azure app registration.",
+				ForceNew:    true,
+			},
+			consts.FieldClientSecret: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "Client Secret of an Azure app registration.",
+				ForceNew:    true,
+			},
+			consts.FieldTenantID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "ID of the target Azure tenant.",
+				ForceNew:    true,
+			},
+			fieldCloud: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies a cloud for the client.",
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func azureSecretsSyncDestinationWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	name := d.Get(consts.FieldName).(string)
+	path := azureSecretsSyncDestinationPath(name)
+
+	data := map[string]interface{}{}
+
+	for _, k := range azureSyncDestinationFields {
+		data[k] = d.Get(k)
+	}
+
+	log.Printf("[DEBUG] Writing Azure sync destination to %q", path)
+	_, err := client.Logical().WriteWithContext(ctx, path, data)
+	if err != nil {
+		return diag.Errorf("error enabling Azure sync destination %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Enabled Azure sync destination %q", path)
+
+	d.SetId(name)
+
+	return azureSecretsSyncDestinationRead(ctx, d, meta)
+}
+
+func azureSecretsSyncDestinationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+	name := d.Id()
+	path := azureSecretsSyncDestinationPath(name)
+
+	log.Printf("[DEBUG] Reading Azure sync destination")
+	resp, err := client.Logical().ReadWithContext(ctx, path)
+	if err != nil {
+		return diag.Errorf("error reading Azure sync destination from %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Read Azure sync destination")
+
+	if resp == nil {
+		log.Printf("[WARN] No info found at %q; removing from state.", path)
+		d.SetId("")
+		return nil
+	}
+
+	if err := d.Set(consts.FieldName, name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, k := range azureSyncDestinationFields {
+		if v, ok := resp.Data[k]; ok {
+			if err := d.Set(k, v); err != nil {
+				return diag.Errorf("error setting state key %q: err=%s", k, err)
+			}
+		}
+	}
+
+	// set sensitive fields that will not be returned from Vault
+	if err := d.Set(consts.FieldClientID, d.Get(consts.FieldClientID).(string)); err != nil {
```

Comment:
While this wouldn't have stopped this from being importable (the `d.Get` commands would just yield empty values), I decided to remove the sensitive field setting regardless. Since the sensitive fields are all required, they would be set to state, so we don't need to explicitly set them from the config